src/main/java/org/folio/helper/FinanceHelper.java

```diff
@@ -621,41 +625,76 @@ private String buildEncumbranceByPolineQuery(String polineId) {
   }
 
   public CompletableFuture<Void> validateExpenseClasses(List<CompositePoLine> poLines) {
-    List<FundDistribution> fundDistributionsWithExpenseClasses = poLines.stream()
+    Map<FundDistribution, List<String>> expenseClassesByFundId = poLines.stream()
       .flatMap(poLine -> poLine.getFundDistribution().stream())
       .filter(fundDistribution -> Objects.nonNull(fundDistribution.getExpenseClassId()))
-      .collect(toList());
+      .collect(groupingBy(Function.identity(), mapping(FundDistribution::getExpenseClassId, toList())));
 
-    return allOf(ctx, fundDistributionsWithExpenseClasses.stream()
+    return allOf(ctx, expenseClassesByFundId.entrySet().stream()
       .map(this::checkExpenseClassIsActiveByFundDistribution)
       .toArray(CompletableFuture[]::new));
   }
 
-  private CompletableFuture<Void> checkExpenseClassIsActiveByFundDistribution(FundDistribution fundDistribution) {
-    String query = String.format("budget.fundId==%s and budget.budgetStatus==Active and status==Inactive and expenseClassId==%s",
-      fundDistribution.getFundId(), fundDistribution.getExpenseClassId());
+  private CompletableFuture<Void> checkExpenseClassIsActiveByFundDistribution(Map.Entry<FundDistribution, List<String>> expenseClassesByFundId) {
```

Comment:
This method might be called while POST `/orders/composite-orders` and PUT `/orders/order-lines/{id}` remember to add modulePermissions there